wolfcrypt/src/asn.c

```diff
@@ -12337,6 +12346,83 @@ static int GenerateDNSEntryIPString(DNS_entry* entry, void* heap)
 }
 #endif /* OPENSSL_ALL || WOLFSSL_IP_ALT_NAME */
 
+#if defined(OPENSSL_ALL)
+/* used to set the human readable string for the registeredID with an
+ * ASN_RID_TYPE DNS entry
+ * return 0 on success
+ */
+static int GenerateDNSEntryRIDString(DNS_entry* entry, void* heap)
+{
+    int i, j, ret = 0;
+    int nameSz;
+    int tmpSize = MAX_OID_SZ;
+    int endChar = 0;
+    int nid     = 0;
+    word32 oid  = 0;
+    word32 idx  = 0;
+    word16 tmpName[MAX_OID_SZ];
+    char finalName[MAX_OID_SZ];
+    char* rid;
+
+    if (entry == NULL || entry->type != ASN_RID_TYPE) {
+        return BAD_FUNC_ARG;
+    }
+
+    if (entry->len <= 0) {
+        return BAD_FUNC_ARG;
+    }
+
+    XMEMSET(&finalName, 0, MAX_OID_SZ);
+    rid = entry->name;
+
+    ret = GetOID((const byte*)rid, &idx, &oid, oidIgnoreType, entry->len);
+
+    if (ret == 0 && (nid = oid2nid(oid, oidCsrAttrType)) > 0) {
+        rid = (char*)wolfSSL_OBJ_nid2ln(nid);
```

Comment:
If wolfSSL_OBJ_nid2ln does not adhere to MAX_OID_SZ then the length of 'rid' will need to be checked before using it with XMEMCPY